src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -173,11 +190,21 @@ public class GenerateProtoTask extends DefaultTask {
     return Integer.MAX_VALUE
   }
 
+  @Optional
+  @Nested
+  DescriptorSetOptions getDescriptorSetOptions() {
+    return generateDescriptorSet ? descriptorSetOptions : null
```

Comment:
If `generateDescriptorSet == false` and this property is accessed, throwing an exception explaining what went wrong would be more helpful.